src/odemis/util/img.py

```diff
@@ -398,20 +404,19 @@ def ensureZYX(data, zIndex=0):
 
     elif len(d.shape) > 2:
 
+        d.shape = d.shape[-3:]  # raise ValueError if it will not work
+        d = d[zIndex]  # Remove z
+
+        # Handle updating metadata
         pxs = d.metadata.get(model.MD_PIXEL_SIZE)
         pos = d.metadata.get(model.MD_POS)
-        
         if pxs is not None and len(pxs) == 3:
```

Comment:
If MD_POS is present, but not PIXEL_SIZE, this will fail, as pxs is `None`, which doesn't support `len()`. => Protect with `...and pxs and len(pxs)==3`.